components/resolved_ts/src/resolver.rs

```diff
@@ -80,6 +81,13 @@ pub struct Resolver {
     locks_by_key: HashMap<Arc<[u8]>, TimeStamp>,
     // start_ts -> locked keys.
     lock_ts_heap: BTreeMap<TimeStamp, TxnLocks>,
+    // the start_ts of large transactions, which use a different tracking strategy with normal
+    // transactions.
+    large_txn_ts: HashSet<TimeStamp>,
+    // each large transaction tracked by this resolver has a representative key tracked. So that
+    // when the large transaction is rolled back, we can rely on this key to guarantee that
+    // eventually there will be no dangling locks.
+    large_txn_key_representative: HashMap<Arc<[u8]>, TimeStamp>,
```

Comment:
> we can rely on this key to guarantee that eventually there will be no dangling locks.

From the code it used to clear the `large_txn_ts` when related locks are untracked, what is the relationship with dangling?


Could the `HashMap<Key, TimeStamp>,` be used instead as there would not be too many representative keys from large transactions?

Reply:
> what is the relationship with dangling?

"orphaned txns" might be more accurate. This is for the case that a large txn entry in resolver is never removed.

> Could the HashMap<Key, TimeStamp>, be used instead

Makes sense. I think `Vec<u8>` is more appropriate here as `Key` is more often for encoded keys.